src/pages/ReimbursementAccount/CompanyStep.js

```diff
@@ -147,10 +196,7 @@ class CompanyStep extends React.Component {
     render() {
         const shouldDisableCompanyName = Boolean(this.props.achData.bankAccountID && this.props.achData.companyName);
         const shouldDisableCompanyTaxID = Boolean(this.props.achData.bankAccountID && this.props.achData.companyTaxID);
-        const missingRequiredFields = this.requiredFields.reduce((acc, curr) => acc || !this.state[curr].trim(), false);
-        const shouldDisableSubmitButton = !this.state.hasNoConnectionToCannabis || missingRequiredFields;
-        const error = this.props.reimbursementAccount.error;
-
+        const shouldDisableSubmitButton = !this.state.hasNoConnectionToCannabis;
```

Comment:
NAB: This seems a bit redundant now that it's simply the inverse of `hasNoConnectionToCannabis`, but I guess it does help add some context

Reply:
FWIW we are talking about never disabling the submit button elsewhere. So this will likely be removed in a future PR. But I agree it's somewhat confusing to have the submit button _only_ disabled for this field. So I think we should either:

- Leave this logic as it was and don't allow submitting if there are missing required fields
- Remove the button disable completely

Reply:
> FWIW we are talking about never disabling the submit button elsewhere. So this will likely be removed in a future PR. But I agree it's somewhat confusing to have the submit button _only_ disabled for this field. So I think we should either:
> 
> * Leave this logic as it was and don't allow submitting if there are missing required fields
> * Remove the button disable completely

I guess I'll disable it completely. Both are a similar amount of work, and If I do this "Leave this logic as it was and don't allow submitting if there are missing required fields", we will have to work again to get to "Remove the button disable completely"

Reply:
I decided in the end to do this: "Leave this logic as it was and don't allow submitting if there are missing required fields". There is more on this on a comment below.